api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -547,4 +547,26 @@ private JSONObject createNotebookCodeCellWithString(String cellInformation) {
         .put("outputs", new JSONArray())
         .put("source", new JSONArray().put(cellInformation));
   }
+
+  @Override
+  public ResponseEntity<EmptyResponse> abortGenomicExtractionJob(
+      String workspaceNamespace, String workspaceId, String jobId) {
+    workspaceAuthService.getWorkspaceEnforceAccessLevelAndSetCdrVersion(
+        workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
+
+    List<GenomicExtractionJob> allWorkspaceExtractJobs =
```

Comment:
Sorry, I missed this the first time through. Why do we need to fetch all of the jobs before we can abort one? Can we just send the abort? (perhaps after we verify that the job exists in our own DB)

If we want the 404, can we just explicitly translate a 404 from Cromwell on abort, and translate this to our own status? (additionally - if the job cannot be aborted due to its status, can we translate that error as well?)

Reply:
Got it - I think an efficient way to do this would be to push this into the `genomicService.abort` call, and change the dao lookup to use both workspace and job ID. This is similar to what Brian did to address the issue in the concept sets controller: https://github.com/all-of-us/workbench/pull/4890

Reply:
Good plan